### ISSUE TITLE
Getting ERR_REQUIRE_ESM

### ISSUE THREAD

**Issue by berkaytheunicorn** [marcomontalbano/figma-export#165]
Hi there, I'm started to get these error when I run figma-export use-confg command.

```

const stripAnsi = require('strip-ansi');
                  ^
Error [ERR_REQUIRE_ESM]: require() of ES Module /Users/XXXXX/node_modules/strip-ansi/index.js from /Users/XXXXX/node_modules/ora/index.js not supported.
Instead change the require of /Users/XXXXX/node_modules/strip-ansi/index.js in /Users/XXXXX/node_modules/ora/index.js to a dynamic import() which is available in all CommonJS modules.
    at Object.<anonymous> (/Users/XXXXX/node_modules/ora/index.js:7:19) {
  code: 'ERR_REQUIRE_ESM'
}

```

My config

```
const fs = require("fs");
const path = require("path");

const customOutput = async (pages) => {
  const output = "./kit/primitives/Icon/resources";

  fs.mkdirSync(output, { recursive: true });

  console.log("OUTPUT", output);

  try {
    const Icons = [];
    pages.forEach(({ name: pageName, components }) => {
      console.log("PAGE", pageName);
      components
        .filter(({ name }) => name.startsWith("Icons/"))
        .forEach(({ name, svg, id }) => {
          const componentName = name.replaceAll("Icons/", "");
          const filePath = path.resolve(output, `${componentName}.tsx`);

          let p = filePath.split("/");
          p.pop();
          p = p.join("/");

          fs.mkdirSync(p, { recursive: true });

          svg = svg
            .replaceAll(`fill-rule`, "fillRule")
            .replaceAll(`clip-rule`, "clipRule")
            .replaceAll(`fill="#121120"`, "")
            .replaceAll(`fill="#000000"`, "")
            .replaceAll(`<svg `, "<svg className={props.className}");

          fs.writeFileSync(
            filePath,
            `
            import React from "react";

            export const Icon${componentName} = (props: { className?:string }) => {
              return (${svg})
            }
          `,
          );
          Icons.push(componentName);
          console.log("GENERATED", componentName);
        });
    });

    const index = `
    import dynamic from "next/dynamic";

    export const RawIcon: Record<string, any> = {
      ${Icons.map(
        (i) =>
          `"${i}": dynamic<any>(() => import("./${i}").then(m=> m.Icon${i})) `,
      ).join(",\n")}
    }

    export type IconName = ${Icons.map((i) => `"${i}"`).join(" | ")}
    `;

    const filePath = path.resolve(output, `index.tsx`);
    fs.writeFileSync(filePath, index);
    console.log("GENERATED", "INDEX");
  } catch (e) {
    console.error(e);
  }
};

module.exports = {
  commands: [
    [
      "components",
      {
        fileId: "xxxx",
        onlyFromPages: ["assets"],
        transformers: [
          require("@figma-export/transform-svg-with-svgo")({
            plugins: ["removeViewBox", "removeDimensions", "cleanupIDs"],
          }),
        ],
        outputters: [customOutput],
      },
    ],
  ],
};
```

> Node.js Version: v20.11.0

**Comment by berkaytheunicorn** [comment]
btw same file works in a different project with different package.json

**Comment by berkaytheunicorn** [comment]
Update: I downgraded to 4.x (globally and locally) now it can work with direct command like `figma-export use-config` but still gives me the same error if I run it with yarn like `yarn export:figma`

**Comment by marcomontalbano** [comment]
Hi @berkaytheunicorn, unfortunately figma-export is not (yet!) an ES module (I planned to do that in v6.x).

I suppose your project has a `type=module` in the package.json. In this case you should follow this piece of documentation "[Node.js ESM](https://github.com/marcomontalbano/figma-export?tab=readme-ov-file#nodejs-esm)". You just simply need to change the figma-export config file extension to `.cjs` and update the script to `figma-export use-config .figmaexportrc.cjs`. In this way, Node.js is able to run the config file as a CommonJS file.

I hope this can help!

**Comment by berkaytheunicorn** [comment]
@marcomontalbano Weirdly I don't have it, i just created new project with create-next-app and move the config file from one of my old projects. That file same version somehow runs perfectly on old project.

**Comment by marcomontalbano** [comment]
**TL;DR** Can you try removing the `yarn.lock` file and `node_modules` folder and run again `yarn install`? After that by running `yarn export:figma` should work :crossed_fingers:

---

I'm able to reproduce the issue following these steps:

```sh
yarn create next-app

# ? What is your project named? › my-app
# ? Would you like to use TypeScript? › Yes
# ? Would you like to use ESLint? › Yes
# ? Would you like to use Tailwind CSS? › Yes
# ? Would you like to use `src/` directory? › No
# ? Would you like to use App Router? (recommended) › Yes
# ? Would you like to customize the default import alias (@/*)? › No

cd my-app/

yarn add\
  @figma-export/cli\
  @figma-export/output-components-as-svg\
  @figma-export/output-styles-as-sass\
  @figma-export/transform-svg-with-svgo
```

Then add the following to the `package.json`:

```diff
  {
    ...
    "scripts": {
      ...,
+     "figma:export": "figma-export use-config"
    },
```

Copy the `.figmaexportrc.js` file that is present in the [README.md](https://github.com/marcomontalbano/figma-export?tab=readme-ov-file#advanced).

By running `yarn figma:export` I get the same issue.

---

It looks like that when Yarn creates the nextjs project and install the figma-export dependencies it resolves the cross-dependencies in a wrong way.
Figma export is using `ora v5.x` that is using `strip-ansi v6.x` under the hood (both are CommonJS library). For some weird reason, when Yarn runs figma-export is trying to use ora, with `string-ansi v7.x` that is an ESM. That's the issue.

By removing the `yarn.lock` file and the `node_modules` folder and running `yarn install`, everything start working as expected. Can you confirm that?

I tried to reproduce the same with `npm create next-app` and `pnpm create next-app`, but everything is working without any other additional changes.

As said I'm planning to move to ESM with figma-export `v6.0`, so this should definitively solve the issue. In the meantime by removing `yarn.lock` and `node_modules` for a blank nextjs project could be a good workaround.

**Comment by berkaytheunicorn** [comment]
Sorry, I just saw your message. I also suspected it was related to yarn, but I couldn't quite pinpoint the issue. Unfortunately, I've already removed the project with this issue because I encountered a similar problem with different libraries as well. In my current setup, I'm not getting any errors and everything works as expected.

**Comment by marcomontalbano** [comment]
That's great, thanks for the confirmation!
